client/src/client/__init__.py

```diff
@@ -115,20 +116,36 @@ def __init__(self) -> None:
                     assert mnemonic and new_passphrase is not None
 
                 self.password = self.startup.get_key(user_passphrase)
-                self.priv_key, self.pub_key = self.startup.load_key_pair(
-                    self.password
-                )
-                self.account = self.startup.load_crypto_account(
-                    user_passphrase
-                )
 
                 # init db manager
-                self.db_manager.open(passphrase=self.password)
+                error: str | None = self.db_manager.open(
+                    passphrase=self.password
+                )
+
+                if error is not None and error == "Wrong password":
+                    feedback = error
+                    continue
 
                 # TODO: some error handling might be useful here
                 self.system = self.db_manager.get_friend(id="system")
                 self.myself = self.db_manager.get_friend(id="myself")
 
+                try:
+                    self.priv_key, self.pub_key = self.startup.load_key_pair(
+                        self.password
+                    )
+                except ValueError:
+                    # Corrupted keys
+                    self._do_graceful_shutdown()
+
+                try:
+                    self.account = self.startup.load_crypto_account(
+                        user_passphrase
+                    )
+                except OlmAccountError as e:
+                    self.log.critical(f"Corrupted OlmAccount: {str(e)}")
+                    self._do_graceful_shutdown()
```

Comment:
Catch all possible startup exceptions in a single catch clause and call graceful shutdown from there. Repeating try/except defeats the purpose of exceptions

Reply:
Done